web/frontend/src/api/slam.ts

```diff
@@ -15,47 +14,9 @@ const concatArrayU8 = (arrays: Uint8Array[]) => {
   return result;
 };
 
-export const getPointCloudMap = (robotClient: Client, name: string) => {
-  const request = new slamApi.GetPointCloudMapRequest();
-  request.setName(name);
-  rcLogConditionally(request);
-
-  const chunks: Uint8Array[] = [];
-  const stream = robotClient.slamService.getPointCloudMap(request);
-
-  stream.on('data', (response) => {
-    const chunk = response.getPointCloudPcdChunk_asU8();
-    chunks.push(chunk);
-  });
-
-  return new Promise<Uint8Array>((resolve, reject) => {
-    stream.on('status', (status) => {
-      if (status.code !== 0) {
-        const error = {
-          message: status.details,
-          code: status.code,
-          metadata: status.metadata,
-        };
-        reject(error);
-      }
-    });
-
-    stream.on('end', (end) => {
-      if (end === undefined) {
-        const error = { message: 'Stream ended without status code' };
-        reject(error);
-      } else if (end.code !== 0) {
-        const error = {
-          message: end.details,
-          code: end.code,
-          metadata: end.metadata,
-        };
-        reject(error);
-      }
-      const arr = concatArrayU8(chunks);
-      resolve(arr);
-    });
-  });
+export const getPointCloudMap = async (slamClient: SlamClient) => {
+  const chunks = await slamClient.getPointCloudMap();
+  return concatArrayU8(chunks);
```

Comment:
Why doesn't the typescript SDK concat the array? This seems like something that shouldn't be left in user-land.

Reply:
I talked to the SLAM team about this, but it seems that this is how it's done in the RDK and other SDK's. It's good practice to match what we have with what's in the RDK.

Reply:
Hmmm anyone you recommend on the SLAM team to sync with concerning this? I disagree with that philosophy - especially here - since most people don't jump languages often and SDKs are often designed around the paradigms and limitations of a particular language. I think that we should be taking advantage of JS as much as possible to offer the smoothest APIs, and that customers will be comparing our SDK to other JS libraries, and not other viam SDKs

Reply:
I edited my original response. I agree with you and have already started making that change in the TS SDK!